Bio/Seq.py

```diff
@@ -78,13 +79,13 @@ class Seq:
     not applicable to protein sequences).
     """
 
-    def __init__(self, data, alphabet=Alphabet.generic_alphabet):
+    def __init__(self, data, alphabet=None):
         """Create a Seq object.
 
         Arguments:
          - seq - Sequence, required (string)
-         - alphabet - Optional argument, an Alphabet object from
-           Bio.Alphabet
+         - alphabet - Optional argument, deprecated. Takes an Alphabet
+           object from Bio.Alphabet, default None.
```

Comment:
"deprecated" is not quite the correct term here. "deprecated" suggests that the function for now still treats alphabets in the same way as before. But in reality we ignore the alphabet altogether. Also I am not sure where an alphabet argument could be coming from, as there won't be an `Alphabet` module any more. So if `alphabet` is not `None`, surely something is wrong, and we should raise a `ValueError`. In any case a `ValueError` is more user-friendly, as the traceback tells you explicitly where the problem is happening, while a `DeprecationWarning` just complains and then gives you the silent treatment.

Reply:
I am giving serious thought to leaving ``Bio.Alphabet`` present (but unused anywhere in Biopython) with a deprecation warning for two releases, so that historic code can continue to do things like ``Seq("aCGT", generic_dna)`` and it will run without changes (just with warnings pushing them to just ``Seq("aCGT")`` instead).

And for those cases where the alphabet was needed for an output format, add a few explicit lines like ``record.annotations["molecule_type"] = "DNA"`` to the code, and it should run on both recent Biopython and Biopython 1.78.

(This won't cover everything, but it would smooth the transition for many users)

Immediate removal forces people to make those code changes immediately if they want to use the latest Biopython.

Reply:
Can you consider the advantages and disadvantages for the user of what you are proposing?

Reply:
That's what I'm trying to do. We might be best writing this down (as a separate issue on GitHub?), starting with some examples that could go into the deprecation file and/or extra chapter in the tutorial. Showing the old style, new style, and a hybrid which could be used if you needed to support both in the short term.

Reply:
Well actually I was implying that the proposal to use a `DeprecatingWarning` has more disadvantages than advantages, and therefore we should not go down that route.

Reply:
Adding a deprecation warning may make writing backward compatible code easier, but it would be ugly - concrete example written up at #3156.

Retaining the stub alphabet argument (even if it must be ``None``) here in the ``Seq`` object and other widely used functions like ``Bio.SeqIO.read`` and ``Bio.SeqIO.parse`` is probably enough. Again, see example.

Reply:
I'll remove the deprecation warning here (and the other classes), leaving a TODO for a ValueError if not None.

Reply:
(I see now that I accidentally included a test commit about the ``__init__`` arguments unrelated to the sequence addition work; removing that commit)